File: java/src/privileges/Delete.java
package privileges;

import decorator.PrivilegeDecorator;
import decorator.User;

public class Delete extends PrivilegeDecorator {
	User user;
	
	public Delete(User user) {
		this.user = user;
	}

	@Override
	public String getPrivileges() {
		return user.getPrivileges() + " delete";
	}
	
	@Override
	public String getName() {
		return user.getName();
	}
	
	
}

File: csharp/decorator_practice/decorator_practice/main/privileges/Delete.cs
using decorator_practice.main.decorator;
using System;
using System.Collections.Generic;
using System.Text;

namespace decorator_practice.main.privileges
{
    public class Delete : PrivilegeDecorator
    {
        /**
	     * User.
	     */
        private User user;

        /**
	     * @param u constructor.
	     */
        public Delete(User u)
        {
            this.user = u;
        }

        /**
	     * @return approve privilege.
	     */
        override
        public String getPrivileges()
        {
            return user.getPrivileges() + " delete";
        }

        /**
	     * @return name.
	     */
        override
        public String getName()
        {
            return user.getName();
        }
    }
}
